solidity/contracts/token/libs/MovableCollateralRouter.sol

```diff
@@ -113,20 +117,45 @@ abstract contract MovableCollateralRouter is FungibleTokenRouter {
         uint256 amount,
         ITokenBridge bridge
     ) external payable onlyRebalancer onlyAllowedBridge(domain, bridge) {
-        address rebalancer = _msgSender();
+        bytes32 recipient = _recipient(domain);
+
+        Quote[] memory quotes = bridge.quoteTransferRemote(
+            domain,
+            recipient,
+            amount
+        );
+
+        uint256 collateralFee = 0;
+        for (uint256 i = 0; i < quotes.length; i++) {
```

Comment:
Gas nit: Cache the length.

Reply:
does this actually save gas? expanding memory by a uint256

Reply:
It should. If you cache you're doing one `mload`, if you don't you're doing `n` `mload`s where `n` is the size of your loop.

Reply:
isnt quotes.length already in memory though?

Reply:
without some docs or minimal repro I'm inclined to leave this for readability